### ISSUE TITLE
Queue position is not saved in the .resume file between sessions

### ISSUE THREAD

**Issue by JP-Ellis** [transmission/transmission#1460]
In my use, I have found that `transmission-daemon` does not remember the queue positions after being restarted.

I have been going through the code, I can't see where the queue position might be save between sessions.  I would have expect this to be in the resume data, but could not find it there.  In my experience, the load order is by `ID` (as per the output of `transmission-remote --list`).

This issue was found with version `3.00 (bb6b5a062e)`.

If this is indeed a missing feature which might be added in the future, I might be able to work on this in the future with a little guidance.

**Comment by ahmar20** [comment]
I think this is related to #1422.

**Comment by JP-Ellis** [comment]
I'm not sure this is related as I'm not getting this error.  The paused/queued status is correctly saved for me.  Also looking at the code, I could not find any place where the queue position might be saved.

**Comment by rezad1393** [comment]
I dont think transmission even saves the queue between runs.
I have the same issue as you.
when I exit transmission-daemon and run it again all the torrents have moved around from their last queue position randomly.

**Comment by afrojewelz** [comment]
yes this feature should be implement somehow , it becames a little more valueable

**Comment by mohkale** [comment]
There was a link to a [forum post](https://forum.transmissionbt.com/viewtopic.php?t=13475) a while back about adding this to transmission but it was discouraging to say the least. 

To summarise: Transmissions download queue is a runtime only queue. It's erased when transmission is stopped and then rebuilt in the same order as torrents are loaded from the filesystem on restart. Torrents are stored by their hash so the queue is always dependent on the hash of the files. That's why transmission generally has a consistent queue ordering across restarts, just not consistent with any sort of ordering you personally create.

Honestly I find this lacking as well. Hopefully if someone submits a PR for it it'll be accepted.

**Comment by rezad1393** [comment]
> There was a link to a [forum post](https://forum.transmissionbt.com/viewtopic.php?t=13475) a while back about adding this to transmission but it was discouraging to say the least.
> 
> To summarise: Transmissions download queue is a runtime only queue. It's erased when transmission is stopped and then rebuilt in the same order as torrents are loaded from the filesystem on restart. Torrents are stored by their hash so the queue is always dependent on the hash of the files. That's why transmission generally has a consistent queue ordering across restarts, just not consistent with any sort of ordering you personally create.
> 
> Honestly I find this lacking as well. Hopefully if someone submits a PR for it it'll be accepted.

I don't even get " consistent queue ordering across restarts,".

**Comment by mohkale** [comment]
@rezad1393 

Huh, that's strange. Maybe it depends on the filesystem or maybe I misunderstood it. Personally I always sort by dateAdded (more predictable and consistent) and I've setup a shell script to re-order the queue based on this.

**Comment by rezad1393** [comment]
I could be wrong and that situation happened before version 3?
I no longer use this app for queue downloads, just regular downloads. So I cant check that anymore.

**Comment by ckerr** [comment]
This is a duplicate of https://github.com/transmission/transmission/issues/155

**Comment by mattpr** [comment]
> @rezad1393
> 
> Personally I always sort by dateAdded (more predictable and consistent) and I've setup a shell script to re-order the queue based on this.

Was looking into this.  Does your shell script use the `transmission-daemon` RPC api or is it just interacting with some GUI app?

As far as I can see there are only a couple mechanisms built into the RPC API that allows changing queue order...

- [torrent-set](https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md#32-torrent-mutator-torrent-set) - args `queuePosition` and `ids` of course...although specifying a fixed queue position with multiple `ids` doesn't make sense to me.
- [queue movement requests](https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md#46-queue-movement-requests), e.g. `queue-move-top` with argument:`ids`.

I've tested both of these with a little bash script with curl with a last-added torrent ID at the bottom of the list (`transmission-remote --list`)...ie the highest torrent ID currently in my list.

It seems to have an effect (torrent goes from "Queued" to "Idle") but the ID doesn't change...so at least in the `transmission-remote --list` view it is still at the bottom of the queue.  So it may also be that the internal queue/priority is disconnected from the IDs.

I feel like I did this before and had a working solution with the RPC to re-order this (changing torrent IDs and position in list view)...but I can't find it.

Anything you can share that I missed is appreciated.

**Comment by mohkale** [comment]
> Was looking into this. Does your shell script use the transmission-daemon RPC api or is it just interacting with some GUI app?

I regret to say it doesn't work. Or perhaps never worked. I did write a script for this a while back. It basically just sent a queue-move-top argument with ids in the order I wanted (this has worked for just moving 5-6 torrents to the top quickly). I assumed it would work for the entire torrent-list but it didn't and I eventually abandoned it. Sorry I couldn't be of more help.

> Does your shell script use the transmission-daemon RPC api or is it just interacting with some GUI app?

I don't use a GUI or transmisson-remote. I use my own python API to communicate with the RPC interface and include it in my [dotfiles](https://github.com/mohkale/dotfiles/tree/master/prog/transmission/cmds/transmission). I also have a script to let me manipulate transmission [using fzf](https://github.com/mohkale/dotfiles/blob/e916bbf95a6b70e259b6cbb073160cc885a86d80/prog/transmission/cmds/tctl).
